### ISSUE TITLE
user tags are not scalable...

### ISSUE THREAD

**Issue by Johnmiicheal** [zcamp-inc/web#13]
Why? cos @Johnmiicheal  thought it was a good idea to send all the usernames across the network to the FE and query the posts if usernames are tagged. 

@owolabioromidayo has a better idea tho...

**Comment by owolabioromidayo** [comment]
i think with the way the post will be stored the only good way now is to do matching on every fetch. storing relations between a post and taggedusers does not really keep ordering and even then we would have to know where they are in the text.

keeping indexes is a thing we can do, and even refresh when posts are edited, but I'm not sure how that translates to how you're transforming the post tags. can you send a screenshot of the tags on the FE? @Johnmiicheal

**Comment by Johnmiicheal** [comment]
![image](https://user-images.githubusercontent.com/69642932/186766526-b16f0227-53d6-4131-904a-43877b6e5c9e.png)

**Comment by owolabioromidayo** [comment]
Oh thats calm then. So it doesnt highlight the @ sign with it?

**Comment by Johnmiicheal** [comment]
wait so does this mean we'll use this??

> So it doesnt highlight the @ sign with it?

No, it doesn't...I'll try to do that tho

**Comment by owolabioromidayo** [comment]
No, the user thing is still not good. I'll find a way to make it more optimal and post specific (maybe a field resolver in the post body that has the same effect but different underpinnings) but that'll require minimal change from your end.

**Comment by Johnmiicheal** [comment]
Aiit
